src/condor_io/condor_auth_passwd.cpp

```diff
@@ -2115,6 +2125,9 @@ Condor_Auth_Passwd::doServerRec2(CondorError* /*errstack*/, bool non_blocking) {
 				}
 				ad.InsertAttr(ATTR_SEC_LIMIT_AUTHORIZATION, ss.str());
 			}
+			ad.InsertAttr(ATTR_TOKEN_SUBJECT, username);
+			ad.InsertAttr(ATTR_TOKEN_ISSUER, issuer);
+			if (!jti.empty()) ad.InsertAttr(ATTR_TOKEN_ID, jti);
```

Comment:
What's the logic behind omitting only `ATTR_TOKEN_ID` if it's empty?

Reply:
The other two attributes are mandatory - it's an implementation bug if this is reachable and they aren't set.  This one is optional (and didn't exist in earlier 8.9.x).

Reply:
If the other two attributes are mandatory, why is it OK for them to be empty?  Lines 2108-2113 explicitly allow for them to be empty; should those lines be changed so that a token without subject or issuer is rejected as malformed?  If we've already done that elsewhere in the code, then why have the conditionals?

Reply:
Actually, I spoke incorrectly above - it's the SciTokens case where the underlying library does this check.

Anyhow, regardless, it seems like a good idea to double-check here to make this explicit.